sdk/helper/storagepacker/ids2keys.go

```diff
@@ -0,0 +1,106 @@
+package storagepacker
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/vault/sdk/helper/cryptoutil"
+	"math"
+	"sort"
+	"strings"
+)
+
+type itemRequest struct {
+	// Item ID, provided by client
+	ID string
+
+	// Storage key == hash of ID
+	Key string
+
+	// Stored object, nil if not found
+	Value *Item
+}
+
+func GetItemIDHash(itemID string) string {
+	return hex.EncodeToString(cryptoutil.Blake2b256Hash(itemID))
+}
+
+// Given a list of IDs, calculate their keys generate itemRequests for each.
+func (s *StoragePackerV2) keysForIDs(ids []string) []*itemRequest {
+	requests := make([]*itemRequest, 0, len(ids))
+	for _, id := range ids {
+		requests = append(requests, &itemRequest{
+			ID:    id,
+			Key:   GetItemIDHash(id),
+			Value: nil,
+		})
+	}
+	return requests
+}
+
+// Given a list of Items, calculate their keys generate itemRequests for each.
+func (s *StoragePackerV2) keysForItems(items []*Item) []*itemRequest {
+	requests := make([]*itemRequest, 0, len(items))
+	for _, i := range items {
+		requests = append(requests, &itemRequest{
+			ID:    i.ID,
+			Key:   GetItemIDHash(i.ID),
+			Value: i,
+		})
+	}
+	return requests
+}
+
+// Sort the requests in key order, nondestructively (so we can refer
+// back to the original order.)
+func sortRequests(requests []*itemRequest) []*itemRequest {
```

Comment:
This looks function looks good and is super easy to follow. Might be not worth the effort, but there is also another common way of implementing support for sorting - basically by implementing the `sort.Interface` [here](https://golang.org/pkg/sort/#Interface) on the item. It ends up getting used [like this](https://gobyexample.com/sorting-by-functions). You might have already seen this but just wanted to surface it in case.

Reply:
Hm, does that have any efficiency gain? I saw that, but it looked like it was a lot more work (three named functions vs. one anonymous one?)

Reply:
I totally agree. It's _maybe_ a more idiomatic approach, but is also more verbose and there's not enough gain to switch over.